src/functional/ffront/foast_to_itir.py

```diff
@@ -160,93 +59,93 @@ def visit_FieldOperator(self, node: foast.FieldOperator, **kwargs) -> itir.Funct
         )
 
     def body_visit(
-        self,
-        exprs: list[foast.Expr],
-        params: Optional[list[itir.Sym]] = None,
-        **kwargs,
-    ) -> itir.Expr:
-        return self.visit(AssignResolver.apply(exprs), **kwargs)
+        self, body: list[foast.Stmt], params: Optional[list[itir.Sym]] = None, **kwargs
+    ) -> itir.FunCall:
+        *assigns, return_stmt = body
+        current_expr = self.visit(return_stmt, **kwargs)
+
+        for assign in assigns[-1::-1]:
```

Comment:
Suggestion for readability:
```suggestion
        for assign in reverse(assigns):
```

Reply:
Will change to `reversed(assigns)`